app/src/main/java/com/example/appmarvel/activity/CharactersListActivity.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.appmarvel.activity
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.example.appmarvel.databinding.ActivityCharactersListBinding
+import com.example.appmarvel.entity.Character
+import com.example.appmarvel.mvvm.viewmodel.CharactersListViewModel
+import com.example.appmarvel.mvvm.viewmodel.CharactersListViewModel.CharactersListState.FETCH_CHARACTERS
+import com.example.appmarvel.mvvm.viewmodel.CharactersListViewModel.CharactersListState.NOT_FOUND_CHARACTERS
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class CharactersListActivity : AppCompatActivity(), KoinComponent {
+
+    private val viewModel: CharactersListViewModel by inject()
+    private lateinit var binding: ActivityCharactersListBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityCharactersListBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        viewModel.fetchCharactersList()
+        viewModel.charactersListState.observe({ lifecycle }, ::changeStateCharacterList)
+    }
+
+    private fun changeStateCharacterList(charactersListData: CharactersListViewModel.CharactersListData) {
+        when (charactersListData.charactersState) {
+            FETCH_CHARACTERS -> showCharactersListOnTextView(charactersListData.data)
+            NOT_FOUND_CHARACTERS -> showErrorMessage(charactersListData?.error)
+        }
+    }
+
+    private fun showCharactersListOnTextView(charactersList: List<Character>) {
+        binding.textViewCharactersActivityCharactersList.text = charactersList.toString()
+    }
+
+    private fun showErrorMessage(error: Exception?) {
+        binding.textViewCharactersActivityCharactersList.text = error.toString()
+    }
+
+    companion object {
+        fun getInstance(context: Context): Intent {
```

Comment:
you can improve this code to be more kotlin style by this

`fun getInstance(context: Context): Intent = Intent(context, CharactersListActivity::class.java)`
or
`fun getInstance(context: Context) = Intent(context, CharactersListActivity::class.java)`